oeps/architectural-decisions/oep-0045/decisions/0002-helm-for-multi-instance-k8s.rst

```diff
@@ -0,0 +1,126 @@
+.. _Helm Charts: https://helm.sh/
+
+
+Consequences
+************
+
+Helm Considerations
+===================
+
+Helm is the best known package manager for Kubernetes. It's status as a graduated project from the Cloud Native Computing Foundation is a clear indication of the thriving adoption, open governance and commitment to the open community.
+
+Helm makes it possible to packages and publish charts, which can be rendered as kubernetes manifests. This charts provide robust support for versioning, deployment and rollback. At the same time Helm makes it easy for operators to customize application configurations during deployment and for developers to provide sensible defaults.
+
+Charts are build with the Go language templating engine which is a drawback since it means adding a new language to the ecosystem with a new learning curve. This is considered acceptable due to the large adoption of Helm in the devops community.
+
+
+Best Practices
+==============
+
+**Cloud Providers**
+
+As much as possible, the Helm chart should aim to be agnostic toward the underlying cloud provider used (e.g. AWS, Azure, Google Cloud Platform, etc.). In cases where provider-specific code is required, the Charts should be written in such a way that they can be configured to work with multiple providers, and not e.g. only support AWS.
+
+**Helm subcharts and modularity**
+
+The project should aim to make the charts available in a way that is composable and granular. Extending a single subchart and using it in the composition should be a better and faster alternative than forking the whole project.
+
+
+Industry Best Practices
+=======================
+
+The recommendations above are heavily inspired by the following resources:
+
+* https://helm.sh/docs/chart_best_practices/
+
+
+Alternatives Considered
+***********************
+
+Building more Kubernetes support into Tutor
+===========================================
+
+Enhancing the current support for tutor core was not considered since it would go against one of the goals of the project as `explained by the author`_.
+
+.. _explained by the author: https://github.com/overhangio/tutor/pull/675#issuecomment-1140919654
+
+Custom Templating tool
+======================
+
+A `custom templating tool`_ built as a tutor plugin and leveraging kustomize for extension was demonstrated and considered as a possible alternative. It was ultimately rejected as Helm would provide more flexibility in extension without the need of maintenance.
+
+.. _custom templating tool: https://github.com/eduNEXT/drydock
+
+
+Implementation Strategy
+***********************
+
+Once this ADR has reached Provisional status:
+
+1. A new repository will be created on GitHub, e.g. at https://github.com/openedx/openedx-helm-charts
+2. The authors of this ADR will create a roadmap using GitHub Issues by defining the initial development issues.
+3. Invite interested parties to become core contributors on the repo, and nominate some of them.
```

Comment:
Should we name the maintainer of the repository in this OEP?

Either way, who will be the maintainer of the repository? :)

Reply:
So far we agreed this will use the  best practices from OEP-55.

This means among other things that we should add a `catalog-info.yaml` file in the repo. Making a PR now.

Reply:
The PR is at https://github.com/openedx/tutor-contrib-multi/pull/16 but it will require an update once the repo changes names.

Does this solve your concern @arbrandes? If so, I'd be ready to merge this PR.

In the meantime I'lll rebase, fix the commit lint check and change the status in the file.

Reply:
@felipemontoya, works for me, thanks!